### PR TITLE
Generate all s2wasm metadata in binaryen

### PR DESCRIPTION
**Pull request by jgravelle-google**
Previously we were parsing the wast in Emscripten to generate these, which 1) is pretty gross, and 2) requires us to emit a wast at all. By moving all that to Binaryen, we can (probably?) skip emitting a wast at all in the near future.

### CODE REVIEW

File: src/wasm-emscripten.cpp [hunk 1]
```diff
@@ -499,15 +507,50 @@ std::string EmscriptenGlueGenerator::generateEmscriptenMetadata(
...
       meta << "\"" << sig << "\"";
     }
     meta << "]";
   }
 
+  meta << ", \"declares\": [";
+  commaFirst = true;
+  for (const auto& import : wasm.imports) {
+    if (import->kind == ExternalKind::Function &&
+        // !import->name.startsWith(EM_JS_PREFIX.str) &&
```
Comment (kripken) [inline src/wasm-emscripten.cpp]: unnecessary comment?

**Comment by jgravelle-google** [comment]
Oh I didn't mention it in the description but this also affects the o2wasm path. Anything wasm-backend. I said "s2wasm" very imprecisely there.

This won't make transitioning any easier (o2wasm path is currently set up with text output to fit here), but it is an improvement to both of them, and some weirdness we can probably remove.

File: src/wasm-emscripten.cpp (hunk 1 above)
Comment (jgravelle-google) [reply src/wasm-emscripten.cpp]: Technically yes. Should probably explicitly annotate this with a `TODO(jgravelle), uncomment when EM_JS lands`. I was thinking one of these would rebase on the other, and fix it then. Explicit is better though.

File: src/wasm-emscripten.cpp [hunk 2]
```diff
@@ -464,32 +464,40 @@ void printSet(std::ostream& o, C& c) {
 std::string EmscriptenGlueGenerator::generateEmscriptenMetadata(
     Address staticBump, std::vector<Name> const& initializerFunctions,
     unsigned numReservedFunctionPointers) {
+  bool commaFirst;
+  auto maybeComma = [&commaFirst](){
```
Comment (aheejin) [inline src/wasm-emscripten.cpp]: nit: whitespace between `()` and `{`?

**Pull request merged**